superset/assets/spec/javascripts/sqllab/LimitControl_spec.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { Label, Popover, Tooltip } from 'react-bootstrap';
+import LimitControl from '../../../src/SqlLab/components/LimitControl';
+
+describe('LimitControl', () => {
+  const defaultProps = {
+    value: 0,
+    defaultQueryLimit: 10000,
+    maxRow: 100000,
+    onChange: () => {},
+  };
+  let wrapper;
+  const factory = o => <LimitControl {...o} />;
+  beforeEach(() => {
+    wrapper = shallow(factory(defaultProps));
+  });
+  it('is a valid element', () => {
+    expect(React.isValidElement(<LimitControl {...defaultProps} />)).toEqual(true);
+  });
+  it('renders a Label', () => {
+    expect(wrapper.find(Label)).toHaveLength(1);
+  });
+  it('loads the correct state', () => {
+    const value = 100;
+    wrapper = shallow(factory({ ...defaultProps, value }));
+    wrapper.find(Label).first().simulate('click');
+    setTimeout(() => {
```

Comment:
setTimeout may cause flaky when running in CI, do you really need it? also I saw in the ListControl you have Overlay, Popover, but you test on Tooltip component?

Reply:
Yeah without `setTimeout` the tests fail.

I replaced `Tooltip` assertions with assertions on `validationErrors` instead.